extensions/rename.md

```diff
@@ -0,0 +1,114 @@
+The final version of the specification will use an unprefixed capability name.
+
+## Introduction
+
+Currently there is no reasonable way to rename channels. All existing methods
+involve closing down the channel and redirecting it to the a new one. This
+is undesirable as access and ban lists needs to be manually copied. It also
+causes problems for client logging as history is split across multiple files.
+
+This specification introduces a method of natively renaming channels that
+does not suffer from these faults.
+
+## Architecture
+
+### Capabilities
+
+This specification introduces the `draft/rename` capability. This capability
+informs the server that the client is capable of handling `RENAME` messages.
+
+### Messages
+
+This specification adds the `RENAME` C2S and S2C message. This message informs
+a client that the name of a channel has been changed.
+
+The `RENAME` message has between two and three parameters. The first parameter
+is the current channel name, the second parameter is the new channel name, and
+the optional third parameter is a reason for renaming the channel.
+
+Server implementations SHOULD NOT allow channels to be converted between types
+with the `RENAME` message. Client implementations SHOULD be able to handle
+renaming between channel types.
+
+### Numerics
+
+| No. | Label               | Format                                         |
+| --- | --------------------| ---------------------------------------------- |
+| XXX | `ERR_CHANNAMEINUSE` | `<nick> <new-channel> :Channel already exists` |
+| XXX | `ERR_CANNOTRENAME`  | `<nick> <old-channel> <new-channel> :<reason>` |
+
+### Examples
+
+Renaming a channel with a reason:
+
+    C: RENAME #boaring #boring :Typo fix
+    S: :nick!user@host RENAME #boaring #boring :Typo fix
+
+Renaming a channel without a reason:
+
+    C: RENAME #coding #knitting
+    S: :nick!user@host RENAME #coding #knitting
+
+Failing to rename a channel because the user does not have the appropriate privileges (this example is non-normative):
+
+    C: RENAME #aniki #aneki
+    S: :irc.example.com 482 nick #aniki :You must be a channel operator
+
+Failing to rename a channel because the new channel name already exists:
+
+    C: RENAME #evil #good :Don't be evil
+    S: :irc.example.com [ERR_CHANNAMEINUSE] nick #good :Channel already exists
+
+Failing to rename a channel because you can not convert between channel types (this example is non-normative):
+
+    C: RENAME #global %local
+    S: :irc.example.com [ERR_CANNOTRENAME] nick #global %local :You cannot change a channel type
+
+Failing to rename a channel because it has been renamed recently:
+
+    C: RENAME #magical-girls #witches
+    S: :irc.example.com [ERR_CANNOTRENAME] nick #magical-girls #witches :This channel has been renamed recently
+
+## Implementation Considerations
+
+Server implementations MUST implement a fallback method for legacy clients.
+This method SHOULD involve sending the legacy client a `PART` message for the
```

Comment:
Should the PART reason should contain the RENAME reason?

Reply:
Yeah, i'll add that.

Reply:
shouldn't it also contain the new channel name?